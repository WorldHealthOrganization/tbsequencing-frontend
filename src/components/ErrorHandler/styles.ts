import { css } from '@emotion/react';

export const styles = {
  wrapper: css({
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    overflow: 'hidden',
  }),
  buttonMargin: css({
    marginTop: '10px',
    zIndex: 999,
  }),
  h1: css({
    textAlign: 'center',
    '&:first-of-type': {
      marginTop: '20px',
    },
  }),
  imageWrapper: css({
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
  }),
  image: css({
    width: '100%',
    bottom: 0,
    position: 'absolute',
  }),
};
