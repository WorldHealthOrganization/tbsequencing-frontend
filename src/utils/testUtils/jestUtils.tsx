import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
