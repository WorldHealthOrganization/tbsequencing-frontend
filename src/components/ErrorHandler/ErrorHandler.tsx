/** @jsxImportSource @emotion/react */
import React from 'react';
import { useNavigate } from 'react-router-dom';
import { styles } from './styles';
import H1 from '../typography/H1';
import imgPath from '../../assets/images/errorSvg.svg';
import { AppButton } from '../AppButton/AppButton';

export const ErrorHandler = () => {
  const navigate = useNavigate();

  const refreshPage = () => {
    navigate(0);
  };

  return (
    <div css={styles.wrapper}>
      <H1 style={styles.h1}>Oops, something went wrong...</H1>
      <H1 style={styles.h1}>
        Please, try again later
      </H1>
      <AppButton style={styles.buttonMargin} onClick={refreshPage} variant="contained" size="medium">Refresh</AppButton>
      <div css={styles.imageWrapper}>
        <img css={styles.image} src={imgPath} alt="error" />
      </div>
    </div>
  );
};

export default ErrorHandler;
