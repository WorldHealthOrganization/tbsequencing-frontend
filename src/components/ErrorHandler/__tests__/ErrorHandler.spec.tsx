import React from 'react';
import { render, getTreeJSONWithRedux, wrappedInRouter } from '../../../utils/testUtils/jestUtils';
import ErrorHandler from '../ErrorHandler';

describe('ErrorHandler component test', () => {
  test('renders ErrorHandler component without error', () => {
    render(
      wrappedInRouter(<ErrorHandler />),
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      wrappedInRouter(<ErrorHandler />),
    );

    expect(tree).toMatchSnapshot();
  });
});
