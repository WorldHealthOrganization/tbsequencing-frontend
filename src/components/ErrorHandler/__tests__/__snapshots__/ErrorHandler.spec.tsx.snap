// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHandler component test Should render correctly 1`] = `
<div
  className="css-l0rjd4-ErrorHandler"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-fgh2mf-MuiTypography-root-BaseText"
  >
    Oops, something went wrong...
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 css-fgh2mf-MuiTypography-root-BaseText"
  >
    Please, try again later
  </p>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-16bw540-MuiButtonBase-root-MuiButton-root-AppButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
    >
      Refresh
    </p>
  </button>
  <div
    className="css-8rl2cb-ErrorHandler"
  >
    <img
      alt="error"
      className="css-1f94h6l-ErrorHandler"
      src="errorSvg.svg"
    />
  </div>
</div>
`;
